packages/tooling/src/oas.js

```diff
@@ -104,6 +104,25 @@ class Operation {
       .map(r => Object.keys(this.responses[r].headers))
       .reduce((a, b) => a.concat(b), []);
 
+    if (includeContent) {
+      // If the operation doesn't already specify a 'content-type' request header,
+      // we check if the path operation request body contains content, which implies that
+      // we should also include the 'content-type' header.
+      if (!this.headers.request.includes('Content-Type') && this.requestBody) {
+        if (this.requestBody.$ref) {
+          const ref = findSchemaDefinition(this.requestBody.$ref, this.oas);
+          if (ref.content && Object.keys(ref.content)) this.headers.request.push('Content-Type');
+        } else if (this.requestBody.content && Object.keys(this.requestBody.content))
+          this.headers.request.push('Content-Type');
+      }
+      // This is a similar approach, but in this case if we check the response content
+      // and prioritize the 'accept' request header and 'content-type' request header
+      if (!this.headers.response.includes('Content-Type') && this.responses) {
+        if (Object.keys(this.responses).some(response => !!this.responses[response].content))
+          this.headers.response.push('Content-Type');
```

Comment:
I'm debating if we should also add an `Accept` request header to this block. We do something similar to that in the explorer: https://github.com/readmeio/api-explorer/blob/d1df9260c6a7d9d106aa277716fb95d56e5a50fb/packages/oas-to-har/src/index.js#L139-L142

Reply:
+1 to adding an Accept header in there if already isn't one.

We should probably replace a lot of the work in oas-to-har with this.

Reply:
Added! https://github.com/readmeio/oas/pull/165/commits/046bd82d5500484b44f919d1355e12ee18ad8ec5